binaries/intra_inter_benchmark.cc

```diff
@@ -70,36 +113,49 @@ int main(int argc, char** argv) {
     at::set_num_threads(FLAGS_intra_op_threads);
   }
 
+  TORCH_CHECK(FLAGS_task_type == "add" || FLAGS_task_type == "mm");
+  run_mm = FLAGS_task_type == "mm";
+
   auto left = at::ones({FLAGS_tensor_dim, FLAGS_tensor_dim}, at::kFloat);
   auto right = at::ones({FLAGS_tensor_dim, FLAGS_tensor_dim}, at::kFloat);
 
-  std::cout << "Launching " << FLAGS_warmup_iter << " warmup tasks" << std::endl;
+  std::cout << "Launching " << pow(2, FLAGS_warmup_iter_pow)
+            << " warmup tasks" << std::endl;
 
   typedef std::chrono::high_resolution_clock clock;
   typedef std::chrono::milliseconds ms;
 
   std::chrono::time_point<clock> start_time = clock::now();
-  launch_tasks_and_wait(left, right, FLAGS_warmup_iter);
+  launch_tasks_and_wait(left, right, FLAGS_warmup_iter_pow);
   auto duration = static_cast<float>(
       std::chrono::duration_cast<ms>(clock::now() - start_time).count());
 
   std::cout << "Warmup time: " << duration << " ms." << std::endl;
 
-  std::cout << "Launching " << FLAGS_iter << " tasks with "
+  std::cout << "Launching " << pow(2, FLAGS_iter_pow) << " tasks with "
             << FLAGS_sub_iter << " subtasks each, using "
             << at::get_num_interop_threads() << " inter-op threads and "
             << at::get_num_threads() << " intra-op threads, "
-            << "tensor dim: " << FLAGS_tensor_dim << std::endl;
+            << "tensor dim: " << FLAGS_tensor_dim
+            << ", task type: " << FLAGS_task_type << std::endl;
 
+  std::vector<float> runtimes;
   for (auto bench_iter = 0; bench_iter < FLAGS_benchmark_iter; ++bench_iter) {
```

Comment:
More things here could be said about clearing the cache etc.

But, btw, why don't we use gbenchmark for all of this?

Reply:
I might change to gbenchmark in the follow ups